### ISSUE TITLE
Unable to load private key

### ISSUE THREAD

**Issue by shaohui-liu2000** [ned1313/Getting-Started-Terraform#8]
I am taking your course on Pluralsight. Under m3 folder, I set up terraform.tfvars as:

aws_access_key = "*******"

aws_secret_key = "******"

key_name = "ssss"

private_key_path = "/Users/myuser/terraform/ssss.pem"

when I do 'ls -l /Users/myuser/terraform/ssss.pem', I got
-rwxr-xr-x 1 myuser  staff  1695 18 Jul 21:48 /Users/myuser/terraform/ssss.pem 

After running 'terraform plan -out m3.tfplan', I got errors when execute 'terraform apply "m3.tfplan"':
_aws_instance.nginx: Destroying... [id=i-06527067970169520]
aws_instance.nginx: Still destroying... [id=i-06527067970169520, 10s elapsed]
aws_instance.nginx: Still destroying... [id=i-06527067970169520, 20s elapsed]
aws_instance.nginx: Still destroying... [id=i-06527067970169520, 30s elapsed]
aws_instance.nginx: Still destroying... [id=i-06527067970169520, 40s elapsed]
aws_instance.nginx: Destruction complete after 41s
aws_instance.nginx: Creating...
aws_instance.nginx: Still creating... [10s elapsed]
aws_instance.nginx: Still creating... [20s elapsed]
aws_instance.nginx: Still creating... [30s elapsed]
aws_instance.nginx: Provisioning with 'remote-exec'..._

_Error: Failed to read ssh private key: no key found_

Can you help this?

My terraform version is 
Terraform v0.12.28
+ provider.aws v2.70.0

and I am on MacOs

**Comment by judavi** [comment]
I'm getting the same error

**Comment by judavi** [comment]
Be careful with the creation of the key. An alternative could be going directly to the AWS website and create the key/pair from there https://us-east-2.console.aws.amazon.com/ec2/v2/home?region=us-east-2#KeyPairs:

**Comment by GitarPlayer** [comment]
my problem was that I created the pair on the CLI copied the private key content to the clipboard and created a new file with vim. Then it had \n (newline characters in the key) when I pasted it into the file. 

@ned1313 Why don't you specify this in the README.md: 
```aws ec2 create-key-pair --key-name MyKeyPair --query 'KeyMaterial' --output text > MyKeyPair.pem```

Just to make it more foolproof ;-)

**Comment by man0l** [comment]
I got the same error.

@GitarPlayer Thank you for your suggestion, it worked for me, but unfortunately, it gave me this error:
 
Error in function call: Call to function "file" failed: contents of /path/to/key/MyKeyPair.pem are not valid UTF-8; use the filebase64 function to obtain the Base64 encoded contents or the other file functions (e.g. filemd5, filesha256) to obtain file hashing results instead.

I have run it in Windows Powershell Terminal and it seems it does not encode the output in UTF-8.

I have converted it by Notepad++ in utf-8 and it worked.
